atc/config.go

```diff
@@ -477,6 +477,15 @@ type PlanConfig struct {
 	Attempts int `json:"attempts,omitempty"`
 
 	Version *VersionConfig `json:"version,omitempty"`
+
+	// name of 'var' step
+	Var string `json:"var,omitempty"`
+
+	// format of input file.
+	Format string `json:"format,omitempty"`
+
+	// if or not dump local vars in 'var' step.
+	Dump bool `json:"dump,omitempty"`
```

Comment:
@vito  I designed the attribute `dump`  is not for post-build debugging purpose, for that, `fly intercept` can be used. Instead, a var might be informational, kinda like metadata of `get/put`, but `task` doesn't have metadata. So `dump` is not only for debugging, but also provide a way for pipeline to display information. From this perspective, `dump` is more descriptive, because it's just dumping the variable value for informational purpose.

I thought to dump var with a better format like:

```
foo.bar = xxx
fool.baa.rr = yyy
```

But it could need more code to write, yeah, I was lazy yesterday :), also I was not sure if you would allow `dump` to go, so I just dumped original content of input file. If you are ok with `dump`, I can change to dump var with above better format.

Reply:
Ahh I see, so it's not really about debugging, it's more about having a way to show interesting/important values in the build output.

I wonder if we could relate this to the redacting behavior? For example, instead of having a flag to enable/disable redacting and a flag to enable/disable dumping of values, we could have a flag that lists which fields to both a) prevent redacting and b) immediately print in the `var` step output in the same way that `get`/`put` metadata is shown:

```yaml
var: branch
reveal: [name]
```

Reply:
I think `reveal` is a good idea. But it would require more changes to store revealed code in database, and GUI code change to show metadata for `var` step. So, let me remove `dump` from this PR, and use a separate PR to add `reveal` to `var` step, it should not be an urgent feature.

Reply:
:+1: Sounds good!

Just an idea: maybe we can add a generic `Metadata` build event which sets the metadata for a given step? We added generic `Initialize/Start/Finish` events with the `set_pipeline` step, but `FinishGet` and `FinishPut` events still exist, with their only difference being that they include the version and metadata (and an `ExitStatus` but that could be simplified to `Succeeded`):

https://github.com/concourse/concourse/blob/4bde6a2d8c77c9cb512c0ec641183112b3019603/atc/event/events.go#L132-L138

Maybe those could be split out into a separate event? Not really sure what to do about the `Version`, since the `var` step won't really need that. Hmm. Maybe it could become an optional step label or something, i.e. `Label: map[string]string`. :thinking: